02-spatial-data.qmd

```diff
@@ -590,8 +590,105 @@ The result `s` is a two-dimensional `numpy` array.
 
 In this section, we are going to demonstrate creation of rasters from scratch. We are going to create two small rasters, `elev` and `grain`, which we are going to use in examples later on in the book. Unlike creating a vector layer, creating a raster from scratch is rarely needed in practive because aligning a raster with the right spatial extent is difficult to do programmatically (GIS software is a better fit for the job). Nevertheless, the examples will be useful to become more familiar with the `rasterio` data structures.
 
-To complete...
+A raster is basically an array combined with georeferencing information, namely:
 
+* A transformation matrix (linking pixel indices with coordinates)
+* A CRS definition
+
+Therefore, to create a raster, we first need to have an array with the values, then supplement it with the georeferencing information. Let's create the arrays `elev` and `grain`. The `elev` array is a 6 by 6 array with sequential values from 1 to 36. It can be created as follows:
```

Comment:
Sentence-based breaks recommended, but we can always tidy up later. The content is most important and that is great!

Reply:
> 



> Sentence-based breaks recommended, but we can always tidy up later. The content is most important and that is great!

Sorry, I'm not used to that yet, will try to keep in mind. Thanks!